lib/services/role.go

```diff
@@ -1034,6 +1038,19 @@ func MatchAzureIdentity(selectors []string, identity string, matchWildcard bool)
 	return false, fmt.Sprintf("no match, role selectors %v, identity: %v", selectors, identity)
 }
 
+// MatchGCPServiceAccount returns true if provided GCP service account matches selectors.
+func MatchGCPServiceAccount(selectors []string, account string, matchWildcard bool) (bool, string) {
```

Comment:
Nit: since as far as I can tell the returned string is only used in tests, could you return a custom integer type instead to avoid allocating strings that won't be used? I guess these strings won't escape to the heap much so it's not a huge deal

Reply:
There is also a variable result in case of failure. I concur there is some performance impact, but it should be small enough not to matter in practice.